### PR TITLE
Simplification-max-zoom; ZoomAnimation and MoveAnimation is same thread now

### PR DESCRIPTION
**Pull request by mikes222**
Simplification-max-zoom implemented as parameter in the map-writer. 
Zoomanimation and movement-animation works now in the same thread.

### CODE REVIEW

**Comment by devemux86** [comment]
>Simplification-max-zoom implemented as parameter in the map-writer.

Useful option. :+1: 

>Zoomanimation and movement-animation works now in the same thread.

Will check this!

BTW it's [recommended](https://help.github.com/articles/about-pull-requests/) using a topic branch for each pull request.
Having that, each feature should be handled in its own pull request for easier merging.

File: mapsforge-map/src/main/java/org/mapsforge/map/model/MapViewPosition.java [hunk 1]
```diff
@@ -250,16 +264,6 @@ public synchronized void init(PreferencesFacade preferencesFacade) {
...
-     * Animates the center position of the map by the given amount of pixels.
-     *
-     * @param moveHorizontal the amount of pixels to move this MapViewPosition horizontally.
-     * @param moveVertical   the amount of pixels to move this MapViewPosition vertically.
-     */
-    public void moveCenter(double moveHorizontal, double moveVertical) {
-        this.moveCenterAndZoom(moveHorizontal, moveVertical, (byte) 0, true);
-    }
-
-    /**
```
Comment (devemux86) [inline mapsforge-map/src/main/java/org/mapsforge/map/model/MapViewPosition.java]: Why different implementation specifically for `moveCenter(moveHorizontal, moveVertical)` ?

**Comment by mikes222** [comment]
Because the original implementation moves the position (with no animation) and starts the zoomanimator with to do (zoomlevel is the same as before). 
The former is a desired behavior the latter not. 

As I recheck the code I would also recommend to change the implementation in 
> public void moveCenter(double moveHorizontal, double moveVertical, boolean animated)

and

>public void setMapPosition(MapPosition mapPosition, boolean animated)

accordingly.

If the animation flag is meant to animate the movement the methods should call animateTo() instead of setCenterInternal().

**Comment by devemux86** [comment]
I think that `animated` parameter was introduced only for zoom.
Nevertheless the internal API could use some improvements.

**Comment by mikes222** [comment]
You are right. It would be better to use the current behavior for existing methods and maybe introduce another family of "animateTo" methods for convenience.

File: mapsforge-map-writer/src/main/java/org/mapsforge/map/writer/MapFileWriter.java [hunk 2]
```diff
@@ -908,14 +908,15 @@ private static long writeSubfile(final long startPositionSubfile, final int zoom
...
-                        LOGGER.info("written " + (processedTiles / amountOfTilesInPercentStep) * PROGRESS_PERCENT_STEP
+                        LOGGER.info("written " + Math.round(processedTiles / amountOfTilesInPercentStep / PROGRESS_PERCENT_STEP * 100)
                                 + "% of sub file for zoom interval index " + zoomIntervalIndex);
                     }
                 }
 
                 // TODO accounting for progress information
             } // end for loop over tile columns
         } // /end for loop over tile rows
+        LOGGER.info("written 100% of sub file for zoom interval index " + zoomIntervalIndex);
```
Comment (devemux86) [inline mapsforge-map-writer/src/main/java/org/mapsforge/map/writer/MapFileWriter.java]: We shouldn't need that, as is causes double logs for 100%.

**Comment by devemux86** [comment]
Thanks!
Merged partially via 70cd87ac7dd31f1a29fc1bdd55bfefb1ab84e8e7 and map writer via dcfa789ceaaa7ce351ec464c5078698523871907.

**Comment by devemux86** [comment]
Merged also `MapViewPosition` via 036e82276d4a9ba00751318da26a4879c6c5badf.
Not included the `moveCenter` change, better seeing that in #893.

**Pull request closed without merging**